src/Functions/seriesOutliersTukey.cpp

```diff
@@ -0,0 +1,173 @@
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
+    {
+        FunctionArgumentDescriptors args{{"time_series", &isArray<IDataType>, nullptr, "Array"}};
+        validateFunctionArgumentTypes(*this, arguments, args);
+
+        return std::make_shared<DataTypeArray>(std::make_shared<DataTypeFloat64>());
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t) const override
+    {
+        ColumnPtr array_ptr = arguments[0].column;
+        const ColumnArray * array = checkAndGetColumn<ColumnArray>(array_ptr.get());
+
+        const IColumn & src_data = array->getData();
+        const ColumnArray::Offsets & src_offsets = array->getOffsets();
+
+        ColumnPtr res;
+
+        if (executeNumber<UInt8>(src_data, src_offsets, res) || executeNumber<UInt16>(src_data, src_offsets, res)
+            || executeNumber<UInt32>(src_data, src_offsets, res) || executeNumber<UInt64>(src_data, src_offsets, res)
+            || executeNumber<Int8>(src_data, src_offsets, res) || executeNumber<Int16>(src_data, src_offsets, res)
+            || executeNumber<Int32>(src_data, src_offsets, res) || executeNumber<Int64>(src_data, src_offsets, res)
+            || executeNumber<Float32>(src_data, src_offsets, res) || executeNumber<Float64>(src_data, src_offsets, res))
+        {
+            return res;
+        }
+        else
+            throw Exception(
+                ErrorCodes::ILLEGAL_COLUMN,
+                "Illegal column {} of first argument of function {}",
+                arguments[0].column->getName(),
+                getName());
+    }
+
+    template <typename T>
+    bool executeNumber(const IColumn & src_data, const ColumnArray::Offsets & src_offsets, ColumnPtr & res_ptr) const
+    {
+        const ColumnVector<T> * src_data_concrete = checkAndGetColumn<ColumnVector<T>>(&src_data);
+        if (!src_data_concrete)
+            return false;
+
+        const PaddedPODArray<T> & src_vec = src_data_concrete->getData();
+
+        auto outliers = ColumnFloat64::create();
+        auto & outlier_data = outliers->getData();
+
+        ColumnArray::ColumnOffsets::MutablePtr res_offsets = ColumnArray::ColumnOffsets::create();
+        auto & res_offsets_data = res_offsets->getData();
+
+        ColumnArray::Offset prev_src_offset = 0;
+        for (auto curr_src_offset : src_offsets)
+        {
+            chassert(prev_src_offset <= curr_src_offset);
+            size_t len = curr_src_offset - prev_src_offset;
+            if (len < 4)
+                throw Exception(ErrorCodes::BAD_ARGUMENTS, "At least four data points are needed for function {}", getName());
+
+            std::vector<Float64> src_sorted(src_vec.begin() + prev_src_offset, src_vec.begin() + curr_src_offset);
```

Comment:
Please move the declaration of `src_sorted` outside the loop. Here, we could use `vector::assign()` to populate it. Right now, the vector is being re-allocated (heap allocation) in each loop iteration, with this proposal, it will retain its capacity between loop iterations and save some allocations.